### PR TITLE
Nb/feat/correlation methods

### PR DESCRIPTION
**Pull request by Nadav-Barak**
Implementation of method as discussed in attached issue.

### CODE REVIEW

File: deepchecks/utils/correlation_methods.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        Returns:
+        --------
+        float
+            Representing the correlation ratio between the features.
+    """
+    fcat, _ = pd.factorize(categorical_feature)
+    cat_num = np.max(fcat) + 1
+    y_avg_array = np.zeros(cat_num)
+    n_array = np.zeros(cat_num)
+    for i in range(0, cat_num):
```
Comment (nirhutnik) [inline deepchecks/utils/correlation_methods.py:100]: no need for 0 in range

File: deepchecks/utils/correlation_methods.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        -----------
+        categorical_feature: Union[List, np.ndarray, pd.Series]
+            A sequence of a categorical feature values without nulls
+        numerical_feature: Union[List, np.ndarray, pd.Series]
+            A sequence of a numerical feature values without nulls
+        Returns:
+        --------
+        float
+            Representing the correlation ratio between the features.
+    """
```
Comment (nirhutnik) [inline deepchecks/utils/correlation_methods.py:95]: If you require values without nulls, why not do a null validation?

File: deepchecks/utils/correlation_methods.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing methods for calculating correlation between features."""
```
Comment (nirhutnik) [inline deepchecks/utils/correlation_methods.py:11]: change usages of "feature" to "variable" - feature is necessarily part of X, but these can be used for correlations with y as well.

Comment (Nadav-Barak) [reply deepchecks/utils/correlation_methods.py:11]: Good sugestion

File: deepchecks/utils/correlation_methods.py (hunk 2 above)
Comment (Nadav-Barak) [reply deepchecks/utils/correlation_methods.py:95]: It is computationally better to it once for the all dataset in the check

File: deepchecks/utils/correlation_methods.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    Returns:
+    --------
+    float
+        Representing the conditional entropy
+    """
+    y_counter = Counter(y)
+    xy_counter = Counter(list(zip(x, y)))
+    total_occurrences = sum(y_counter.values())
+    s_xy = 0.0
+    for xy in xy_counter:
```
Comment (noamzbr) [inline deepchecks/utils/correlation_methods.py:42]: Can this be vectorized?

**Review by noamzbr — APPROVED** [review]

**Pull request merged**